analyzers/tests/SonarAnalyzer.UnitTest/TestFramework/EnvironmentalVariableScope.cs

```diff
@@ -0,0 +1,84 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2021 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+
+namespace SonarAnalyzer.UnitTest.Helpers
+{
+    /// <summary>
+    /// Defines a scope inside which new environment variables can be set.
+    /// The variables will be cleared when the scope is disposed.
+    /// Copy-pasted from S4NET codebase.
+    /// </summary>
+    public sealed class EnvironmentVariableScope : IDisposable
+    {
+        private IDictionary<string, string> originalValues = new Dictionary<string, string>();
+
+        public void SetVariable(string name, string value)
```

Comment:
`SetVariable` vs `SetVariableInRelease` confuses me from API/usage point of view.
When using the class (long after I forget this detail), I will want to have just a single method that will do the trick for me. And that will deal with the specific context for me so I don't have to think about it everytime I use this.

I suggest to introduce constructor with optional argument that will be explicitly set only for those specific tests and that will do this trick for us by default otherwise.